master/internal/api/handler.go

```diff
@@ -39,14 +39,14 @@ func Route(handler func(c echo.Context) (interface{}, error)) echo.HandlerFunc {
 }
 
 // WebSocketRoute upgrades incoming requests to websocket requests.
-func WebSocketRoute(handler func(socket *websocket.Conn, c echo.Context) error) echo.HandlerFunc {
+func WebSocketRoute(handler func(socket ReadWriter, c echo.Context) error) echo.HandlerFunc {
```

Comment:
I'm thinking that since this is a webscoket-specific (io-specific) function, it makes sense to me to leave it alone, and transition from websocket to the interface in PublisherSet.Websocket().

For instance, I don't see WebSocketRoute in the integration test, right?  So it doesn't really need to change at all.